### ISSUE TITLE
Link List

### ISSUE THREAD

**Issue by SteveP-UK** [OpenLiveWriter/OpenLiveWriter#100]
When I installed OLW, most settings from WLW imported fine except for my 'link list'. Is there a way of importing it manually?

**Comment by ScottIsAFool** [comment]
1. Do Windows Key + R (run command), then paste in this: %appdata%\Windows Live Writer
2. Copy the LinkGlossary folder
3. Go up one folder level
4. Open the Open Live Writer folder
5. Paste the LinkGlossary folder in here (overwrite if needed)

This should give you all the items you used to have in WLW.

**Comment by SteveP-UK** [comment]
Hi Scott

Brilliant! Thank you so much for the quick reply.

Thanks for all your future help!

Regards

Steve
www.StevePalmer.com
@TweetsBySteveP
+44 (0)1536 312 766
07860 639661

**Comment by hashhar** [comment]
Can I try and implement an "Import LinkList from WLW" button for this?

**Comment by ScottIsAFool** [comment]
Ideally we probably want to create a list of things that aren't going to be migrated across automatically. I wrote a backup utility (http://wlwbackup.codeplex.com/) back in the day that could probably now do with some love and some migration options adding that would take all of this into consideration already (it backs up link glossaries)

**Comment by hashhar** [comment]
See if I got this right:
1. We'll present the user with a list of things that won't be migrated during the installation only if we detect an existing install of WLW.
2. Offer them a link to your tool to migrate those things too.

**Comment by ScottIsAFool** [comment]
I wouldn't offer a link to my tool. I think there could be a better discussion about whether we want to offer a migration from WLW > OLW from writer itself. @RobDolinMS @willduff @shanselman thoughts? Migration would likely include blog theme files etc as well as link glossary.

**Comment by hashhar** [comment]
I will take a look at your tool and try to understand how it works so that its features may be merged.

**Comment by ScottIsAFool** [comment]
The tool itself does nothing more than take a copy of certain folders in appdata and the plugins folder, as well as make a backup of the WLW registry then just zip them all up :)

**Comment by SteveP-UK** [comment]
Cheers again Scott. I've always found your WLW backup utility 'back in the
day' to be mighty useful when migrating WLW from PC to PC

Regards

Steve
www.StevePalmer.com
blog.StevePalmer.com
@TweetsBySteveP
Mobile: 07860 639 661
:::::::::::::::::::::::::::::::::::::::::::::::::::

**Comment by hashhar** [comment]
Oh. It'll be easy then I guess. Just a matter of dropping those files into the correct directories and updating relevant configs.

**Comment by timheuer** [comment]
I actually wouldn't be in favor if implementing these migration things **in** the OLW product itself.  Given they would be ephemeral features for the user (one-time operation) it would ideally be a) a part of the setup/install options or b) just a separate quick tool we could point people to.

I'm in favor of b) to not introduce a whole wizard/UI step into the OLW base that we'd eventually want to remove at some point.

My 2 cents.

**Comment by ScottIsAFool** [comment]
Fair enough. So I definitely need to give my backup utility some love then :)

**Comment by hashhar** [comment]
I was talking about the setup itself. The first time that the user installs it, we should detect if they have a WLW installation and then ask them if they want to import or not.

**Comment by SteveP-UK** [comment]
Yes agreed, from my point of view as just an enthusiastic end-user, I feel
it should be a separate tool/part of setup routine - in theory, I'll only
ever use it once, or when I'm migrating from one PC to another.

Just my 2 pence worth!

Regards

Steve
www.StevePalmer.com
@TweetsBySteveP
+44 (0)1536 312 766
07860 639661

**Comment by willduff** [comment]
Thinking about how I'd approach this, my idea would be to package @ScottIsAFool's backup/migration tool into a NuGet package, publish that, make it a dependency for Writer and package it with our installer. On first launch of Open Live Writer, we could offer to launch the backup/migration tool (or call into the DLL directly and run it immediately). We'd ship this for v1.0 and then rip it out sometime later.

**Comment by SteveP-UK** [comment]
For what it's worth, I've used @ScottIsAFool's backup utility for restoring
various settings when migrating WLW to other PCs and it's always been very
reliable.

Regards

Steve

**Comment by hashhar** [comment]
I agree with @willduff 's approach. It should be a part of the installation process (or the first launch?).

**Comment by hashhar** [comment]
I can see a function in `OpenLiveWriter.ApplicationMain.cs` called `MaybeMigrateSettings()` which checks the `SettingsVersion` property in the registry and copies the entire key and subkeys to the `OpenLiveWriter` key.

We can tap into this function and make it copy the `%appdata%\Windows Live Writer` contents over to `%appdata%\OpenLiveWriter`.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
